In [9]:
#EJERCICIO1
import pandas as pd
df = pd.read_csv('clima_talca_enero2024.csv')
print("Primeras 5 filas")
display(df.head(5))
print("Últimas 3 filas")
display(df.tail(3))
print("Información del DataFrame")
df.info()
print("Estadísticas descriptivas")
display(df.describe())
print("Filas y columnas")
display(df.shape)

Primeras 5 filas


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
0,2024-01-01,32.1,18.2,45.0,0.0,Soleado
1,2024-01-02,33.5,17.5,48.0,0.0,Soleado
2,2024-01-03,NaN,19.0,42.0,0.0,Parcial
3,2024-01-04,31.8,NaN,50.0,0.0,Soleado
4,2024-01-05,34.2,17.8,NaN,0.0,Soleado


Últimas 3 filas


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
28,2024-01-29,31.4,17.2,49.0,0.0,Parcial
29,2024-01-30,30.7,16.9,51.0,0.0,Nublado
30,2024-01-31,29.9,17.4,53.0,0.0,Soleado


Información del DataFrame
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31 entries, 0 to 30
Data columns (total 6 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   fecha             31 non-null     object 
 1   temp_max          25 non-null     float64
 2   temp_min          25 non-null     float64
 3   humedad_pct       28 non-null     float64
 4   precipitacion_mm  28 non-null     float64
 5   condicion         29 non-null     object 
dtypes: float64(4), object(2)
memory usage: 1.6+ KB
Estadísticas descriptivas


,temp_max,temp_min,humedad_pct,precipitacion_mm
count,25.000000,25.00000,28.000000,28.000000
mean,32.236000,17.84000,49.107143,0.321429
std,1.936896,0.87892,4.433369,0.840792
min,28.800000,16.20000,42.000000,0.000000
25%,30.700000,17.20000,45.750000,0.000000
50%,32.100000,17.80000,49.000000,0.000000
75%,33.700000,18.50000,52.250000,0.000000
max,35.400000,19.30000,58.000000,3.200000


Filas y columnas


(31, 6)

In [10]:
#EJERCICIO2
nulos_por_columna = df.isnull().sum()
print("\n conteo de nulos por columna")
print(nulos_por_columna)

porcentaje_nulos = (nulos_por_columna / len(df)) * 100
print("\n porcentaje de nulos por columna")
print(porcentaje_nulos)

columnas_mas_15 = porcentaje_nulos[porcentaje_nulos > 15]
print("\n columnas con más del 15% de nulos")
print(columnas_mas_15)


 conteo de nulos por columna
fecha               0
temp_max            6
temp_min            6
humedad_pct         3
precipitacion_mm    3
condicion           2
dtype: int64

 porcentaje de nulos por columna
fecha                0.000000
temp_max            19.354839
temp_min            19.354839
humedad_pct          9.677419
precipitacion_mm     9.677419
condicion            6.451613
dtype: float64

 columnas con más del 15% de nulos
temp_max    19.354839
temp_min    19.354839
dtype: float64


In [11]:
#EJERCICIO3
df_limpio = df.copy()

df_limpio['temp_max'] = df_limpio['temp_max'].fillna(df_limpio['temp_max'].mean())
df_limpio['temp_min'] = df_limpio['temp_min'].fillna(df_limpio['temp_min'].mean())

df_limpio['humedad_pct'] = df_limpio['humedad_pct'].fillna(df_limpio['humedad_pct'].median())

df_limpio['precipitacion_mm'] = df_limpio['precipitacion_mm'].fillna(0.0)

df_limpio = df_limpio.dropna(subset=['condicion'])

print("nulos despues de la limpieza")
print(df_limpio.isnull().sum())


nulos despues de la limpieza
fecha               0
temp_max            0
temp_min            0
humedad_pct         0
precipitacion_mm    0
condicion           0
dtype: int64


In [12]:
#EJERCICIO4
display(df_limpio[['fecha', 'temp_max', 'condicion']])

display(df_limpio.iloc[:5])

display(df_limpio.loc[14])

display(df_limpio[df_limpio['temp_max'] >= 34])

display(df_limpio[df_limpio['condicion'] == 'Lluvioso'])

display(df_limpio[(df_limpio['temp_max'] > 30) & (df_limpio['humedad_pct'] > 50)])

df_limpio['amplitud_termica'] = df_limpio['temp_max'] - df_limpio['temp_min']
display(df_limpio.head())

,fecha,temp_max,condicion
0,2024-01-01,32.100,Soleado
1,2024-01-02,33.500,Soleado
2,2024-01-03,32.236,Parcial
3,2024-01-04,31.800,Soleado
4,2024-01-05,34.200,Soleado
5,2024-01-06,35.000,Caluroso
6,2024-01-07,32.236,Caluroso
7,2024-01-08,33.100,Parcial
8,2024-01-09,30.500,Nublado
9,2024-01-10,29.800,Lluvioso


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
0,2024-01-01,32.100,18.20,45.0,0.0,Soleado
1,2024-01-02,33.500,17.50,48.0,0.0,Soleado
2,2024-01-03,32.236,19.00,42.0,0.0,Parcial
3,2024-01-04,31.800,17.84,50.0,0.0,Soleado
4,2024-01-05,34.200,17.80,49.0,0.0,Soleado


fecha               2024-01-15
temp_max                  35.4
temp_min                  19.1
humedad_pct               43.0
precipitacion_mm           0.0
condicion             Caluroso
Name: 14, dtype: object

,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
4,2024-01-05,34.2,17.8,49.0,0.0,Soleado
5,2024-01-06,35.0,18.5,44.0,0.0,Caluroso
13,2024-01-14,34.9,18.7,49.0,0.0,Soleado
14,2024-01-15,35.4,19.1,43.0,0.0,Caluroso
24,2024-01-25,34.0,17.6,47.0,0.0,Soleado
25,2024-01-26,35.1,18.9,44.0,0.0,Caluroso


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
9,2024-01-10,29.800,17.1,53.0,1.8,Lluvioso
10,2024-01-11,31.200,18.3,49.0,0.0,Lluvioso
20,2024-01-21,28.800,16.2,58.0,0.0,Lluvioso
21,2024-01-22,32.236,17.0,55.0,3.2,Lluvioso


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion
7,2024-01-08,33.100,17.84,51.0,0.0,Parcial
8,2024-01-09,30.500,16.50,55.0,2.5,Nublado
17,2024-01-18,31.000,17.30,52.0,0.0,Parcial
18,2024-01-19,30.200,16.80,54.0,0.0,Nublado
21,2024-01-22,32.236,17.00,55.0,3.2,Lluvioso
22,2024-01-23,31.500,17.84,52.0,1.5,Parcial
29,2024-01-30,30.700,16.90,51.0,0.0,Nublado


,fecha,temp_max,temp_min,humedad_pct,precipitacion_mm,condicion,amplitud_termica
0,2024-01-01,32.100,18.20,45.0,0.0,Soleado,13.900
1,2024-01-02,33.500,17.50,48.0,0.0,Soleado,16.000
2,2024-01-03,32.236,19.00,42.0,0.0,Parcial,13.236
3,2024-01-04,31.800,17.84,50.0,0.0,Soleado,13.960
4,2024-01-05,34.200,17.80,49.0,0.0,Soleado,16.400


In [13]:
#EJERCICIO5
# 1. Día más caluroso del mes
dia_mas_caluroso = df_limpio.loc[df_limpio['temp_max'].idxmax()]
print("--- Día más caluroso ---")
print(dia_mas_caluroso[['fecha', 'temp_max']])

# 2. Día con mayor precipitación
dia_mayor_lluvia = df_limpio.loc[df_limpio['precipitacion_mm'].idxmax()]
print("\n--- Día con mayor precipitación ---")
print(dia_mayor_lluvia[['fecha', 'precipitacion_mm']])

# 3. Conteo de días para cada condición climática
print("\n--- Frecuencia de condiciones climáticas ---")
print(df_limpio['condicion'].value_counts())

# 4. Temperatura máxima y mínima promedio del mes
print("\n--- Temperaturas promedio del mes ---")
print(f"Máxima promedio: {df_limpio['temp_max'].mean():.2f}")
print(f"Mínima promedio: {df_limpio['temp_min'].mean():.2f}")

# 5. Días duplicados en el dataset
duplicados = df_limpio.duplicated().sum()
print(f"\n--- Días duplicados encontrados: {duplicados} ---")

# 6. Resumen final con describe() solo de las columnas numéricas
print("\n--- Resumen descriptivo final ---")
display(df_limpio.describe())

--- Día más caluroso ---
fecha       2024-01-15
temp_max          35.4
Name: 14, dtype: object

--- Día con mayor precipitación ---
fecha               2024-01-22
precipitacion_mm           3.2
Name: 21, dtype: object

--- Frecuencia de condiciones climáticas ---
condicion
Soleado     9
Parcial     6
Caluroso    6
Nublado     4
Lluvioso    4
Name: count, dtype: int64

--- Temperaturas promedio del mes ---
Máxima promedio: 32.20
Mínima promedio: 17.84

--- Días duplicados encontrados: 0 ---

--- Resumen descriptivo final ---


,temp_max,temp_min,humedad_pct,precipitacion_mm,amplitud_termica
count,29.000000,29.000000,29.000000,29.000000,29.000000
mean,32.204000,17.840000,49.172414,0.310345,14.364000
std,1.788428,0.813722,4.310030,0.827796,1.404882
min,28.800000,16.200000,42.000000,0.000000,12.000000
25%,31.000000,17.300000,46.000000,0.000000,13.236000
50%,32.236000,17.840000,49.000000,0.000000,14.000000
75%,33.500000,18.300000,52.000000,0.000000,15.860000
max,35.400000,19.300000,58.000000,3.200000,16.500000


### Respuestas - Ejercicio 1

**P1.1 ¿Cuántas filas y columnas tiene el dataset?**
El dataset tiene 31 filas y 6 columnas.

**P1.2 ¿Qué columnas contienen datos de tipo float? ¿Cuál es la temperatura máxima promedio según describe()?**
Las columnas float son: `temp_max`, `temp_min`, `humedad_pct` y `precipitacion_mm`. La temperatura máxima promedio es de 32.24°C.

**P1.3 ¿Qué información relevante entrega df.info() que NO entrega df.describe()?**
Entrega los tipos de datos, el recuento de valores no nulos (útil para ver faltantes) y el uso de memoria.

---

### Respuestas - Ejercicio 2

**P2.1 ¿Cuántos valores faltantes tiene el dataset en total?**
Tiene un total de 20 valores faltantes.

**P2.2 ¿Qué columnas superan el 15% de nulos? ¿Qué implicancia tiene esto para el análisis?**
Las columnas `temp_max` y `temp_min` superan el 15%. Esto implica que no podemos simplemente borrar esas filas, debemos rellenar los datos (imputarlos) para no perder el 40% del dataset.

**P2.3 ¿Por qué es importante calcular el porcentaje y no sólo el conteo absoluto de nulos?**
Porque el porcentaje indica el impacto real del problema. Faltar 10 datos en 30 filas es grave, pero 10 en un millón es irrelevante.

---

### Respuestas - Ejercicio 3

**P3.1 ¿Por qué se usó fillna(mean()) para temp_max y fillna(median()) para humedad_pct?**
La media es mejor para datos estables (temperaturas). La mediana es ideal cuando hay saltos bruscos o valores atípicos que alterarían el promedio (humedad).

**P3.2 ¿Cuántas filas se eliminaron al aplicar dropna(subset=['condicion'])? ¿Es aceptable esta pérdida?**
Se eliminaron 2 filas. Es aceptable porque representa solo un ~6% de pérdida de datos.

**P3.3 ¿Qué sucede si se aplica df.dropna() sin parámetros?**
Eliminaría cualquier fila con al menos un nulo. Quedarían solo 16 filas, perdiendo casi la mitad de los datos del mes.

---

### Respuestas - Ejercicio 4

**P4.1 ¿Cuántos días tuvo una temperatura máxima igual o superior a 34°C? ¿Qué condición climática predominó esos días?**
Fueron 6 días. Predominaron las condiciones "Caluroso" y "Soleado".

**P4.2 ¿Cuál es la amplitud térmica promedio del mes? ¿Qué día tuvo la mayor amplitud?**
La amplitud térmica promedio fue de ~14.5°C. La mayor amplitud fue el 6 de enero (16.5°C).

**P4.3 ¿Por qué se utilizan paréntesis adicionales en el filtro compuesto?**
Porque en Pandas los operadores lógicos (`&`, `|`) tienen mayor prioridad. Sin paréntesis, Python se confunde y arroja error.

---

### Respuestas - Ejercicio 5

**P5.1 ¿Qué condición climática dominó en enero 2024 en Talca? ¿Es consistente con el clima real de la zona en verano?**
Dominó "Soleado". Sí es consistente, ya que el clima mediterráneo de la zona en verano es cálido y seco.

**P5.2 ¿La temperatura mínima promedio es consistente con las temperaturas nocturnas de Talca en enero?**
Sí. Talca tiene una alta amplitud térmica; días muy calurosos, pero noches lo suficientemente frescas (aprox. 17.8°C promedio).

**P5.3 Si necesitaras compartir este análisis con alguien sin conocimientos de Python, ¿cómo presentarías los resultados?**
Omitiría el código y mostraría la información con gráficos visuales simples, priorizando los hallazgos principales (días extremos, promedios y distribución del clima).


